src/core/include/cesium/omniverse/ObjectPool.h

```diff
@@ -84,8 +84,12 @@ template <typename T> class ObjectPool {
     virtual std::shared_ptr<T> createObject(uint64_t objectId) = 0;
     virtual void setActive(std::shared_ptr<T> object, bool active) = 0;
 
+    const std::deque<std::shared_ptr<T>>& getQueue() {
+        return _queue;
+    }
+
   private:
-    std::queue<std::shared_ptr<T>> _queue;
+    std::deque<std::shared_ptr<T>> _queue;
```

Comment:
This double-ended data structure is still only being used as a queue. Is it faster for setting capacity or something?

Reply:
Interesting. An `std::queue` is just a wrapper around `std::deque`. Then by def I imagine `std::queue` has to be at least some small amount slower.

Reply:
The switch is actually for a different reason, `std::queue` doesn't support iterators and therefore doesn't support range based for loops which I needed to do here:

```
void FabricMaterialPool::updateShaderInput(const pxr::SdfPath& shaderPath, const pxr::TfToken& attributeName) {
    const auto& fabricMaterials = getQueue();
    for (auto& fabricMaterial : fabricMaterials) {
        fabricMaterial->updateShaderInput(
            FabricUtil::toFabricPath(shaderPath), FabricUtil::toFabricToken(attributeName));
    }
}
```